src/app/child-dev-project/notes/model/note.ts

```diff
@@ -15,26 +15,38 @@
  *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { MeetingNoteAttendance } from "../meeting-note-attendance";
 import { DatabaseEntity } from "../../../core/entity/database-entity.decorator";
 import { Entity } from "../../../core/entity/entity";
 import { DatabaseField } from "../../../core/entity/database-field.decorator";
 import { WarningLevel, WarningLevelColor } from "../../warning-level";
 import { InteractionType } from "../note-config-loader/note-config.interface";
+import { EventAttendance } from "../../attendance/model/event-attendance";
+import { AttendanceStatus } from "../../attendance/model/attendance-status";
+import { v4 as uuid } from "uuid";
 
 @DatabaseEntity("Note")
 export class Note extends Entity {
+  static create(date: Date, subject: string = ""): Note {
+    const instance = new Note(uuid());
+    instance.date = date;
+    instance.subject = subject;
+    return instance;
+  }
+
   /** IDs of Child entities linked with this note */
   @DatabaseField() children: string[] = [];
-  /** An array of triplets containing information about the child and it's attendance */
-  @DatabaseField() attendances: MeetingNoteAttendance[] = [];
+  /** optional additional information about attendance at this event for each of the linked children */
+  @DatabaseField() private childrenAttendance: Map<
+    string,
+    EventAttendance
+  > = new Map();
```

Comment:
Did you already check if this format is correctly stored on the database or if we need a special datatype definition for this?

Reply:
thanks for pointing this out! having bad unit tests is not helpful at all ...

please check and merge PR #573 also as a basis for this